### PR TITLE
Add X-Wing combiner

### PR DESCRIPTION
**Pull request by raphaelrobert**
Adds [draft-connolly-cfrg-xwing-kem-01](https://www.ietf.org/archive/id/draft-connolly-cfrg-xwing-kem-01.html).

### CODE REVIEW

File: src/kem.rs [hunk 1]
```diff
@@ -147,13 +153,44 @@ impl X25519MlKem768Draft00PrivateKey {
     }
 }
 
+/// An X-Wing private key.
+pub struct XWingKemDraft01PrivateKey {
```
Comment (franziskuskiefer) [inline src/kem.rs:157]: There's a small diff to -02. Can you update to that?

File: src/kem.rs [hunk 2]
```diff
@@ -183,13 +220,41 @@ impl X25519MlKem768Draft00PublicKey {
     }
 }
 
+/// An X-Wing public key.
+pub struct XWingKemDraft01PublicKey {
+    pub pk_m: MlKem768PublicKey,
+    pub pk_x: x25519::PublicKey,
+}
+
+// TODO: Check serialization
```
Comment (franziskuskiefer) [inline src/kem.rs:229]: Is this something you still want to do?

File: src/kem.rs [hunk 3]
```diff
@@ -445,6 +565,18 @@ impl Ss {
                 out.extend_from_slice(kk.as_ref());
                 out
             }
+            Ss::XWingKemDraft01(ss_m, ss_x, ct_x, pk_x) => {
+                let x1 = r"\./";
```
Comment (franziskuskiefer) [inline src/kem.rs:569]: This can be updated "In hex XWingLabel is given by 5c2e2f2f5e5c."

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, generally looks good with a few nits.

File: src/kem.rs (hunk 2 above)
Comment (raphaelrobert) [reply src/kem.rs:229]: This was already addressed

Comment (raphaelrobert) [reply src/kem.rs]: This was already addressed

**Pull request merged**